SpokestackTests/SpeechContextTest.swift

```diff
@@ -0,0 +1,233 @@
+        XCTAssert(l.didDidInit)
+        
+        // start
+        l.reset()
+        let didStartExpectation = expectation(description: "didStartExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.didStart.")
+        l.asyncExpectation = didStartExpectation
+        c.notifyListener(.start)
+        wait(for: [didStartExpectation], timeout: 1)
+        XCTAssert(l.didDidStart)
+
+        // stop
+        l.reset()
+        let didStopExpectation = expectation(description: "didStopExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.didStop.")
+        l.asyncExpectation = didStopExpectation
+        c.notifyListener(.stop)
+        wait(for: [didStopExpectation], timeout: 1)
+        XCTAssert(l.didDidStop)
+
+        // Activate
+        l.reset()
+        let didActivateExpectation = expectation(description: "didActivateExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.Activate.")
+        l.asyncExpectation = didActivateExpectation
+        c.notifyListener(.activate)
+        wait(for: [didActivateExpectation], timeout: 1)
+        XCTAssert(l.activated)
+
+        // dectivate
+        l.reset()
+        let didDeactivateExpectation = expectation(description: "didDeactivateExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.dectivate.")
+        c.setListener(l)
+        l.asyncExpectation = didDeactivateExpectation
+        c.notifyListener(.deactivate)
+        wait(for: [didDeactivateExpectation], timeout: 1)
+        XCTAssert(l.deactivated)
+        
+        // recognize
+        l.reset()
+        let didRecognizeExpectation = expectation(description: "didRecognizeExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.recognize.")
+        l.asyncExpectation = didRecognizeExpectation
+        c.notifyListener(.recognize)
+        wait(for: [didRecognizeExpectation], timeout: 1)
+        
+        // timeout
+        l.reset()
+        let didTimeoutExpectation = expectation(description: "didTimeoutExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.timeout.")
+        l.asyncExpectation = didTimeoutExpectation
+        c.notifyListener(.timeout)
+        wait(for: [didTimeoutExpectation], timeout: 1)
+
+        // error
+        l.reset()
+        // no error set still sends the event
+        let didErrorErrorExpectation = expectation(description: "didErrorErrorExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.error.")
+        l.asyncExpectation = didErrorErrorExpectation
+        c.notifyListener(.error)
+        wait(for: [didErrorErrorExpectation], timeout: 1)
+        // error set sends event
+        let didErrorExpectation = expectation(description: "didErrorExpectation fulfills when testNotifyListener calls SpeechContextTestDelegate.error.")
+        l.asyncExpectation = didErrorExpectation
+        c.notifyListener(.error)
```

Comment:
Doesn't look like you're actually setting an error here.

Reply:
Yeah, it's actually testing the guard clause fallback, need to add a happy path test too.